hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1907,6 +1908,19 @@ private static Double castToDouble(Object val) {
 
   public static boolean isColumnTypeSupported(Schema schema, Option<HoodieRecordType> recordType) {
     Schema schemaToCheck = resolveNullableSchema(schema);
+    // Check for precision and scale if the schema has a logical decimal type.
+    LogicalType logicalType = schemaToCheck.getLogicalType();
+    if (logicalType != null && logicalType instanceof LogicalTypes.Decimal) {
+      LogicalTypes.Decimal decimalType = (LogicalTypes.Decimal) logicalType;
+      // The maximum allowed precision and scale as per the payload schema. See DecimalWrapper in HoodieMetadata.avsc:
+      // https://github.com/apache/hudi/blob/45dedd819e56e521148bde51a3dfa4e472ea70cd/hudi-common/src/main/avro/HoodieMetadata.avsc#L247
+      final int maxPrecision = 30;
+      final int maxScale = 15;
+      if (decimalType.getPrecision() > maxPrecision || decimalType.getScale() > maxScale) {
```

Comment:
The logic here is not correct. You need `decimalType.getPrecision() + (maxScale - decimalType.getScale())` > maxPrecision || decimalType.getScale() > maxScale)` so you account for the change in precision when the scale is updated.

Reply:
thanks, i did not account for upscaling. Updated the condition.